### PR TITLE
[SPARK-32852][SQL][Doc][Followup] Revise the documentation of spark.sql.hive.metastore.jars

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

This is a follow-up for https://github.com/apache/spark/pull/29881.
It revises the documentation of the configuration `spark.sql.hive.metastore.jars`.

### Why are the changes needed?

Fix grammatical error in the doc. 
Also, make it more clear that the configuration is effective only when `spark.sql.hive.metastore.jars` is set as `path`

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Just doc changes.

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @AngersZhuuuu @cloud-fan

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala [hunk 1]
```diff
@@ -95,18 +95,19 @@ private[spark] object HiveUtils extends Logging {
...
-      s" 1. file://path/to/jar/xxx.jar\n" +
-      s" 2. hdfs://nameservice/path/to/jar/xxx.jar\n" +
-      s" 3. /path/to/jar/ (path without URI scheme follow conf `fs.defaultFS`'s URI schema)\n" +
-      s" 4. [http/https/ftp]://path/to/jar/xxx.jar\n" +
-      s"Notice: `http/https/ftp` doesn't support wildcard, but other URLs support" +
-      s"nested path wildcard, Such as: " +
-      s" 1. file://path/to/jar/*, file://path/to/jar/*/*\n" +
-      s" 2. hdfs://nameservice/path/to/jar/*, hdfs://nameservice/path/to/jar/*/*\n" +
-      s"When ${HIVE_METASTORE_JARS.key} is set to `path`, we will use Hive jars configured by this")
+    val HIVE_METASTORE_JARS_PATH = buildStaticConf("spark.sql.hive.metastore.jars.path")
```
Comment (dongjoon-hyun) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala:98]: indentation?

Comment (gengliangwang) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala:98]: Thanks, updated

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala [hunk 2]
```diff
@@ -96,17 +96,18 @@ private[spark] object HiveUtils extends Logging {
...
-      s" 2. hdfs://nameservice/path/to/jar/xxx.jar\n" +
-      s" 3. /path/to/jar/ (path without URI scheme follow conf `fs.defaultFS`'s URI schema)\n" +
-      s" 4. [http/https/ftp]://path/to/jar/xxx.jar\n" +
-      s"Notice: `http/https/ftp` doesn't support wildcard, but other URLs support" +
-      s"nested path wildcard, Such as: " +
-      s" 1. file://path/to/jar/*, file://path/to/jar/*/*\n" +
-      s" 2. hdfs://nameservice/path/to/jar/*, hdfs://nameservice/path/to/jar/*/*\n" +
-      s"When ${HIVE_METASTORE_JARS.key} is set to `path`, we will use Hive jars configured by this")
+    .doc(s"""
+      | Comma-sperated paths of the jars that should be used to instantiate the HiveMetastoreClient.
```
Comment (dongjoon-hyun) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala:100]: Could you revert from `sperated` (this) to `separated` (original)?

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala [hunk 3]
```diff
@@ -96,17 +96,18 @@ private[spark] object HiveUtils extends Logging {
...
+      | This configuration is effective only when `{$HIVE_METASTORE_JARS.key}` is set as `path`.
+      | The paths can be any of the following format:
+      | 1. file://path/to/jar/foo.jar
+      | 2. hdfs://nameservice/path/to/jar/foo.jar
+      | 3. /path/to/jar/ (path without URI scheme follow conf `fs.defaultFS`'s URI schema)
+      | 4. [http/https/ftp]://path/to/jar/foo.jar
+      | All the above formats support wildcard except for the `http/https/ftp` URL. For example:
```
Comment (dongjoon-hyun) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala:107]: Maybe, the following is shorter?
```
- All the above formats support wildcard except for the `http/https/ftp` URL. For example:
+ 1, 2, and 3 support wildcard. For example:
```

**Review by dongjoon-hyun — APPROVED** [review]

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala (hunk 2 above)
Comment (gengliangwang) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala]: Thanks, updated

**Review by dongjoon-hyun — APPROVED** [review]

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala [hunk 4]
```diff
@@ -96,17 +96,18 @@ private[spark] object HiveUtils extends Logging {
...
+      | This configuration is effective only when `{$HIVE_METASTORE_JARS.key}` is set as `path`.
+      | The paths can be any of the following format:
+      | 1. file://path/to/jar/foo.jar
+      | 2. hdfs://nameservice/path/to/jar/foo.jar
+      | 3. /path/to/jar/ (path without URI scheme follow conf `fs.defaultFS`'s URI schema)
+      | 4. [http/https/ftp]://path/to/jar/foo.jar
+      | Note that 1, 2, and 3 support wildcard. For example:
+      | 1. file://path/to/jar/*,file://path2/to/jar/*/*.jar
+      | 2. hdfs://nameservice/path/to/jar/*,hdfs://nameservice2/path/to/jar/*/*.jar
+      """.stripMargin)
```
Comment (AngersZhuuuu) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala:110]: LGTM

**Review by AngersZhuuuu — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by gengliangwang** [comment]
@dongjoon-hyun @AngersZhuuuu @cloud Thanks for the review

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**